File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder("<");
        String[] inputStr = br.readLine().split(" ");
        int N = Integer.parseInt(inputStr[0]);
        int K = Integer.parseInt(inputStr[1]);

        Deque<Integer> deque = new LinkedList<>();
        for (int i = 0; i < N; i++) {
            deque.offer(i + 1);
        }

        while (!deque.isEmpty()) {
            for (int i = 1; i < K; i++) {
                deque.offer(deque.pollFirst());
            }

            if (deque.size() != 1) {
                sb.append(deque.pollFirst()).append(", ");
            } else {
                sb.append(deque.pollFirst());
            }
        }

        System.out.println(sb.append(">"));
    }
}

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val (n, k) = readLine().split(" ").map { it.toInt() }
    val deque: ArrayDeque<Int> = ArrayDeque()
    for (i in 1..n) {
        deque.addLast(i)
    }

    val sb = StringBuilder("<")
    while (deque.isNotEmpty()) {
        repeat(k - 1) {
            deque.addLast(deque.removeFirst())
        }

        sb.append(deque.removeFirst())
        if (deque.isNotEmpty()) {
            sb.append(", ")
        }
    }
    sb.append(">")

    print(sb)
}